unison-cli/src/Unison/Codebase/Editor/HandleCommand.hs

```diff
@@ -93,10 +108,11 @@ commandLine
   -> (Int -> IO gen)
   -> Free (Command IO i v) a
   -> IO a
-commandLine config awaitInput setBranchRef rt notifyUser notifyNumbered loadSource codebase serverBaseUrl rngGen =
- flip State.evalStateT 0 . Free.fold go
+commandLine config awaitInput setBranchRef rt notifyUser notifyNumbered loadSource codebase serverBaseUrl rngGen free = do
+ v <- STM.newTVarIO 0
+ flip runReaderT v . Free.fold go $ free
  where
-  go :: forall x . Command IO i v x -> State.StateT Int IO x
+  go :: forall x . Command IO i v x -> ReaderT (STM.TVar Int) IO x
```

Comment:
I changed StateT into ReaderT so that I can implement MonadUnliftIO over the result monad.